src/common/meson.build

```diff
@@ -0,0 +1,384 @@
+        description: 'Define to 1 if the GNU C extensions __builtin_bswap16/32/64 are supported.'
+        )
+else
+    have_byteswap_h = false
+    have_sys_endian_h = false
+    have_sys_byteorder_h = false
+    bswap_headers = [
+        'byteswap.h',
+        'sys/endian.h',
+        'sys/byteorder.h'
+    ]
+    foreach h : bswap_headers
+        if cc.has_header(h)
+            config_h.set10('HAVE_' + h.underscorify().to_upper(), true)
+            set_variable(f'have_' + h.underscorify(), true)
+        endif
+    endforeach
+
+    if have_byteswap_h
+        bswap_fcts = [
+            'bswap_16',
+            'bswap_32',
+            'bswap_64'
+        ]
+        foreach f : bswap_fcts
+            bswap_src = '''
+#include <byteswap.h>
+int main()
+{
+	@0@(42);
+	return 0;
+}
+'''
+            if cc.links(bswap_src.format(f), name: f)
+                config_h.set10('HAVE_' + f.to_upper(), true,
+                    description: 'Define to 1 if ' + f + '() is available.'
+                )
+            endif
+        endforeach
+    endif
+endif
+
+fast_unaligned_access = false
+if get_option('unaligned-access').auto()
+    unaligned_access_cpus = [
+        'ppc',
+        'ppc64',
+        'x86',
+        'x86_64',
+    ]
+    if host_machine.cpu_family() in unaligned_access_cpus
+        config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+            description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+        )
+        fast_unaligned_access = true
+    endif
+
+    if not fast_unaligned_access
+        unaligned_access_cpus = [
+            'aarch64',
+            'arm',
+            'riscv32',
+            'riscv64',
+        ]
+        if host_machine.cpu_family() in unaligned_access_cpus
+            ua_src = '''
+#if !defined(__ARM_FEATURE_UNALIGNED) \
+		&& !defined(__riscv_misaligned_fast) \
+		&& !defined(_MSC_VER)
+compile error
+#endif
+int main(void) { return 0; }
+'''
+            if cc.compiles(ua_src, name: 'fast unaligned access for ' + host_machine.cpu_family())
+                config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+                description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+                )
+                fast_unaligned_access = true
+            endif
+        endif
+    endif
+elif get_option('unaligned-access').enabled()
+    config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+        description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+    )
+    fast_unaligned_access = true
+endif
+
+if get_option('unsafe-type-punning')
+    config_h.set10('HAVE___BUILTIN_ASSUME_ALIGNED', true,
+        description: 'Define to 1 if the GNU C extension __builtin_assume_aligned is supported.'
+                    )
+endif
+
+# tuklib_mbstr.m4
+
+### FIXME : mbrtowc is supported in C99 spec and on Windows
```

Comment:
This is true. When I added the check, I felt it was the simplest way to ensure that the code will keep working on odd platforms. [Gnulib's mbrtowc](https://www.gnu.org/software/gnulib/manual/html_node/mbrtowc.html) lists _old versions_ of Minix 3 and HP-UX and also IRIX as lacking `mbrtowc`. I don't think those matter in Meson build at all. The latest MINIX 3 does have `mbrtowc` and so does HP-UX. IRIX is too old.

So it's possible that, among the supported or potentially-supported platforms, only DJGPP lacks `mbrtowc`. If so, it could be handled with a special `#ifdef` for that in the C code. But it's also not much work to check it in the build system.

I currently believe it's fine if the Meson build unconditionally defines `HAVE_MBRTOWC`. It might be cleaned up later along with `HAVE_STDINT_H` and such.

Reply:
done in f51a11f845a064f8410dd1cb3411dfd0344d0dad